### PR TITLE
Add swift 5.7 nightly

### PR DESCRIPTION
**Pull request by igor-makarov**
Together with https://github.com/compiler-explorer/compiler-explorer/pull/3764

### CODE REVIEW

File: bin/yaml/swift.yaml [hunk 1]
```diff
@@ -31,6 +31,17 @@ compilers:
           - '5.4'
           - '5.5'
           - '5.6'
+    5.7-nightly:
+      if: nightly
+      type: restQueryTarballs
+      install_always: true
+      url: https://swift.org/builds/swift-5.7-branch/ubuntu2004/latest-build.yml
```
Comment (mattgodbolt) [inline bin/yaml/swift.yaml:38]: Oh wow! Top marks for working out how to do that in my shonky configuration language!

File: bin/yaml/swift.yaml [hunk 2]
```diff
@@ -31,6 +31,17 @@ compilers:
...
+      query: |
+        'https://swift.org/builds/swift-5.7-branch/ubuntu2004/' \
```
Comment (mattgodbolt) [inline bin/yaml/swift.yaml:40]: you might be able to put `{name}` in place of the `{5.7}` here to make it a little more honest...I'll merge and check if that works ok! Thanks

**Review by mattgodbolt — APPROVED** [review]

File: bin/yaml/swift.yaml (hunk 2 above)
Comment (igor-makarov) [reply bin/yaml/swift.yaml:40]: This def might be also useful for later, when a new stable branch appears and 5.7 is released officially, correct?

Comment (mattgodbolt) [reply bin/yaml/swift.yaml:40]: Exactly! In this instance (I made the change and tested it) - we can just add a new `target: - 5.8` below and we're golden. Assuming nothing else changes :)

Thanks for the patch, I tested it locally. It'll install tonight -- I'm busy this weekend else I'd promise to get this and the other change deployed; maybe one of the other admins will get to it! Thanks for your contribution!

File: bin/yaml/swift.yaml (hunk 1 above)
Comment (igor-makarov) [reply bin/yaml/swift.yaml:38]: Hehe, it's not so shonky in comparison to others I've encountered 😂

File: bin/yaml/swift.yaml (hunk 2 above)
Comment (igor-makarov) [reply bin/yaml/swift.yaml:40]: Thank you!

**Pull request merged**